### PR TITLE
[feature/#460] Set NavigationRail when display expanded or device is tablet

### PR DESCRIPTION
**Pull request by l2hyunwoo**
## Overview (Required)
- Phase 1: Apply Navigation Rail when foldable device expanded(or device is tablet)
- I think UI test should be added to next PR, what do you think about it? 
  - If UI test is essential when ui is modified, I'll include UI test code in this.

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by takahirom** [comment]
Looks great. Could you add test something like this?

@Config(qualifiers = RobolectricDeviceQualifiers.MediumTablet)
...
@Test
fun checkMediumTabletLaunchShot()...

**Comment by l2hyunwoo** [comment]
> Looks great. Could you add test something like this?
> ```
> @Config(qualifiers = RobolectricDeviceQualifiers.MediumTablet)
> ...
> @Test
> fun checkMediumTabletLaunchShot()...
> ```

It's in new commit!

File: feature/main/src/main/java/io/github/droidkaigi/confsched2023/main/MainScreen.kt [hunk 1]
```diff
@@ -63,22 +78,36 @@ interface MainNestedGraphStateHolder {
...
 }
 
+enum class NavigationType {
+    BOTTOM_NAVIGATION, NAVIGATION_RAIL
+}
+
 @Composable
 fun MainScreen(
+    windowSize: WindowSizeClass,
+    displayFeatures: ImmutableList<DisplayFeature>,
```
Comment (takahirom) [inline feature/main/src/main/java/io/github/droidkaigi/confsched2023/main/MainScreen.kt:88]: What do you think you are going to use this? 👀

File: feature/main/src/main/java/io/github/droidkaigi/confsched2023/main/MainScreen.kt [hunk 2]
```diff
@@ -137,34 +166,49 @@ data class MainScreenUiState(
...
-            startDestination = "timetable",
-            modifier = Modifier.padding(padding),
-            enterTransition = { materialFadeThroughIn() },
-            exitTransition = { materialFadeThroughOut() },
-        ) {
-            mainNestedNavGraph(mainNestedNavController, padding)
+        }
+        Scaffold(
+            bottomBar = {
+                AnimatedVisibility(visible = navigationType == BOTTOM_NAVIGATION) {
```
Comment (takahirom) [inline feature/main/src/main/java/io/github/droidkaigi/confsched2023/main/MainScreen.kt:190]: 👍

File: feature/main/src/main/java/io/github/droidkaigi/confsched2023/main/MainScreen.kt (hunk 1 above)
Comment (l2hyunwoo) [reply feature/main/src/main/java/io/github/droidkaigi/confsched2023/main/MainScreen.kt:88]: 1) windowSize: to calculate WindowWidthSizeClass and apply proper navigationType(in this pr) and content pane type(in next PR, apply dual-pane(as known as List-Detail Pattern) or single-pane)

2) displayFeatures: When WindowWidthSizeClass is Medium, content arrangement type should be determined by device folding posture. (according to material 3 sample application [reply](https://github.com/android/compose-samples/blob/main/Reply/app/src/main/java/com/example/reply/ui/ReplyApp.kt#L99))

Comment (l2hyunwoo) [reply feature/main/src/main/java/io/github/droidkaigi/confsched2023/main/MainScreen.kt:88]: As [explained](https://github.com/DroidKaigi/conference-app-2023/issues/460#issuecomment-1684697989), list-detail pattern will apply in this screen. So `displayFeatures` will be used to implement it.

Comment (takahirom) [reply feature/main/src/main/java/io/github/droidkaigi/confsched2023/main/MainScreen.kt:88]: Great! Thanks

**Review by takahirom — APPROVED** [review]
Looks awesome!

**Pull request merged**